Helper/Order.php

```diff
@@ -701,14 +688,60 @@ public function formatReferenceUrl($reference)
     }
 
     /**
-     * Format the (class internal) unambiguous transaction state out of the type and status
+     * Format the (class internal) unambiguous transaction state out of the type and status.
+     * Infer eventually missing states. Represent Bolt partial capture AUTHORIZED state as CAPTURED.
      *
      * @param \stdClass $transaction
+     * @param OrderPaymentInterface $payment
      * @return string
      */
-    public function getTransactionState($transaction)
+    public function getTransactionState($transaction, $payment)
     {
-        return $transaction->type.":".$transaction->status;
+        $transactionState = $transaction->type.":".$transaction->status;
+        $prevTransactionState = $payment->getAdditionalInformation('transaction_state');
+        $prevCaptures = $payment->getAdditionalInformation('captures') ?: [];
+
+        if (!$prevTransactionState) {
+            if (in_array($transactionState, [self::TS_ZERO_AMOUNT, self::TS_CREDIT_COMPLETED])) {
+                return $transactionState;
+            }
+            return self::TS_PENDING;
+        }
+
+        if (in_array($prevTransactionState, [self::TS_PENDING, self::TS_REJECTED_REVERSIBLE]) &&
+            $transactionState == self::TS_AUTHORIZED &&
+            $transaction->captures
+        ) {
+            return self::TS_AUTHORIZED;
```

Comment:
comment above says `Represent Bolt partial capture AUTHORIZED state as CAPTURED` but here you use `authorized`?

Reply:
@daisy1754 
This is the case when, for some reason, the base AUTHORIZED state was not recorded, and there has already been a partial capture. We first record the AUTHORIZED event and then the CAPTURE on the next hook.